bin/alias.ml

```diff
@@ -79,14 +79,38 @@ let dep_on_alias_rec_multi_contexts ~dir:src_dir ~name ~contexts =
   let* dir =
     Action_builder.of_memo (find_dir_specified_on_command_line ~dir:src_dir)
   in
-  let+ is_nonempty_list =
+  let* is_nonempty_list =
     Action_builder.all
       (List.map contexts ~f:(fun ctx ->
            Action_builder.dep_on_alias_rec name ctx dir))
   in
+  let+ load_dir =
+    Action_builder.all
+    @@ List.map contexts ~f:(fun ctx ->
+           let dir =
+             Source_tree.Dir.path dir
+             |> Path.Build.append_source (Context_name.build_dir ctx)
+             |> Path.build
+           in
+           Action_builder.of_memo @@ Load_rules.load_dir ~dir)
+  in
   let is_nonempty = List.exists is_nonempty_list ~f:Fun.id in
   if (not is_nonempty) && not (Dune_engine.Alias.is_standard name) then
-    User_error.raise
+    let hints =
+      let rec aliases t =
+        match (t : Load_rules.Loaded.t list) with
+        | Build build :: t ->
+          Alias.Name.Map.union ~f:(fun _ _ _ -> None) build.aliases (aliases t)
```

Comment:
I'm really not sure about my choice for ~f here. I just wanted to take the union of the map.

Reply:
returning `None` will remove the key when it's present in both maps, I don't think you want that. Taking either value and returning with `Some` should work.

Reply:
OK done.